Create 1.24.0 release documentation (#212)
canonical/anbox-cloud-docs: reference/release-notes/1.24.0.md

## New features & improvements

### Updates to components

* Anbox Cloud includes and uses the following latest versions for components it uses:
  - LXC version 6.0.2<!--AC-2812-->
  - Mesa driver version 24.2.4<!--AC-2800-->
  - VirGL renderer version 1.9.0.<!--AC-2800-->

### Security

* We have enhanced security with a minimum version of TLS 1.3 as a requirement for all services working with Anbox Cloud. TLS 1.3 is also enabled by default for the Anbox Stream Agent and Gateway.<!--AC-2728 and AC-2654-->

### Anbox Management Service (AMS)

* We have increased the maximum number of queued operations in AMS to allow better performance.<!--AC-2831-->
* AMS supports querying the image variant (`android` or `aaos`) for Anbox Cloud images.<!--AC-2757-->
* The web client, for example, the Anbox Cloud dashboard can communicate with multiple AMS instances.<!--AC-2674-->

### Android

* This release includes Android 14 based AOSP and AAOS images.
* We now support using the lock screen in Anbox Cloud([LP 1983426](https://bugs.launchpad.net/anbox-cloud/+bug/1983426)).

> * The out of band v2 feature in Anbox Cloud requires the Android app to be running to receive broadcasts. If the app is in the [cached state](https://developer.android.com/guide/components/activities/process-lifecycle), the system places [context-registered broadcasts in a queue]((https://developer.android.com/develop/background-work/background-tasks/broadcasts#android-14)) and hence the app may not receive broadcasts immediately as it would when it is running.<!--AC-2852-->

Reviewer: The out of band v2 feature in Anbox Cloud requires the Android app to be running to receive broadcasts when using the Android 14 LXD image. In short, this is Android 14 specific.
That implies if people are using Android 12 or Android 13 LXD images, apps can stay cache state and the out of band v2 feature worked fine.

* The Anbox Streaming SDK now includes two additional examples that demonstrate shader-based upscaling algorithms integrated with the JS SDK:<!--AC-2543-->
  - [Snapdragon™ Game Super Resolution(SGSR)](https://github.com/SnapdragonStudios/snapdragon-gsr)
  - [Anime4K high-quality real-time anime](https://github.com/bloc97/Anime4K)

### Other

* Android security updates for November 2024 (see [Android Security Bulletin - November 2024](https://source.android.com/docs/security/bulletin/2024-11-01) for more information).
* The Android WebView has been updated to 130.0.6723.73.

## Removed functionality

Following deprecation in earlier releases, these functionalities have been removed from Anbox Cloud with the 1.24.0 release:

* Support for Ubuntu 20.04 (Focal Fossa)

## Deprecations

> < List deprecated features. Provide an alternative, if any, for the users to use instead of the deprecated features >

Author: I wouldn't do it that way because it switches the context of the release notes which is to focus on the current release. So any deprecations we are announcing in 1.24.0 will go here. 

I have added a statement that, in addition to calling out no deprecations for 1.24.0, will remind users to look at our deprecation notices to prepare for next releases.

## Known issues

* Out of band support does not work as expected for Android 14 images because of a change in implicit intents feature in the upstream.<!--AC-2852-->

> * The following features can be enabled for AAOS images based on Android 14. However, this is not supported and may result in unexpected behaviour. This will be removed in the next patch release:<!--AC-2848-->

Reviewer 2: As in, saying that we share the same source tree this time around? Is it useful for users to know?

> * The following features can be enabled for AAOS images based on Android 14. However, this is not supported and may result in unexpected behaviour. This will be removed in the next patch release:<!--AC-2848-->

Author: It is a technical implementation detail, no? I don't think this will be useful.

  - `enable_virtual_keyboard`
  - `enable_system_ui`
  - `enable_anbox_ime`
  - Custom Android ID
  - Custom system apps

>   - Boot package support

Author: Done, thanks!

## Requirements

See the [Requirements](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/requirements/) for details on general and deployment specific requirements to run Anbox Cloud.

## New features & improvements

### Updates to components

* Anbox Cloud includes and uses the following latest versions for components it uses:
  - LXC version 6.0.2<!--AC-2812-->
  - Mesa driver version 24.2.6<!--AC-2800-->
  - VirGL renderer version 1.9.0.<!--AC-2800-->

### Security

* We have enhanced security with a minimum version of TLS 1.3 as a requirement for all services working with Anbox Cloud. TLS 1.3 is also enabled by default for the Anbox Stream Agent and Gateway.<!--AC-2728 and AC-2654-->

### Anbox Management Service (AMS)

* We have increased the maximum number of queued operations in AMS to allow better performance.<!--AC-2831-->
* Anbox Cloud provides the image type (`android`, `aaos` or `generic`) through the following three ways:<!--AC-2757-->
  - `metadata.yaml` of the Anbox Cloud image
  - API response from the AMS when querying image information
  - Output of `amc image show` command

>   This allows the web client to adapt its interface according to the image type.

Author: Not relevant as the feature is rewritten without these details

> * We have enabled support for multiple cluster management by AMS.<!--AC-2674-->

Reviewer: This is not accurate.

The story is about enabling dashboard to communicate with multiple AMS instances.
However when I looked at the AC-2674, I didn't find a PR yet. So I am not quite sure if this feature is enabled or not in this cycle.
@jat-canonical  could you clarify on this as well?
Thanks

---
orphan: true
---
# 1.24.0

These release notes cover new features and changes in Anbox Cloud 1.24.0.

Anbox Cloud 1.24.0 is a minor release. To understand minor and patch releases, see [Release notes](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/release-notes/release-notes).

Please see [Component versions](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/component-versions/) for a list of updated components.

## Requirements

See the [Requirements](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/requirements/) for details on general and deployment specific requirements to run Anbox Cloud.

## New features & improvements

### Updates to components

* Anbox Cloud includes and uses the following latest versions for components it uses:
  - LXC version 6.0.2<!--AC-2812-->

>   - Mesa driver version 24.2.6<!--AC-2800-->

Author: Need confirmation on Mesa driver version as Jira story just says latest version, I took 24.2.6 from the upstream but is that the one we are on?

Reviewer 2: We're running 24.2.4 in our case, I just checked
```suggestion
  - Mesa driver version 24.2.4<!--AC-2800-->
```

  - VirGL renderer version 1.9.0.<!--AC-2800-->

### Security

* We have enhanced security with a minimum version of TLS 1.3 as a requirement for all services working with Anbox Cloud. TLS 1.3 is also enabled by default for the Anbox Stream Agent and Gateway.<!--AC-2728 and AC-2654-->

### Anbox Management Service (AMS)
* We have increased the maximum number of queued operations in AMS to allow better performance.<!--AC-2831-->

> * Anbox Cloud provides the image type (`android`, `aaos` or `generic`) through the following three ways:<!--AC-2757-->

Reviewer 2: I think `generic` is not something we officially support for users, and more of an internal thing for now.
@adglkh @morphis Wdyt?

Author: I have removed generic from the image type now but for the generic application streaming, let's confirm with @morphis

>   - `metadata.yaml` of the Anbox Cloud image

Reviewer 2: Do users need to know this? We don't have to hide it, but it may not be too useful for them to know this anyway (as they probably cannot change it easily).

Author: Thanks for the clarification

>   - API response from the AMS when querying image information

Reviewer 2: Should we add a link to the API docs here?

Author: Not relevant as the section is rewritten

  - Output of `amc image show` command

  This allows the web client to adapt its interface according to the image type.
* We have enabled support for multiple cluster management by AMS.<!--AC-2674-->

### Android

> * This release includes Android 14 based AOSP images.

Reviewer 2: And AAOS:
```suggestion
* This release includes Android 14 based AOSP and AAOS images.
```

* We now support using the lock screen in Anbox Cloud([LP 1983426](https://bugs.launchpad.net/anbox-cloud/+bug/1983426)).

### Charms

* For the NFS Operator charm for Anbox Cloud, `Cachefilesd` is now optional and is used only when `fsc` is provided as a mount option.<!--AC-2839-->

### Streaming improvements

* Anbox Cloud now provides an option to allow the use of custom fragment shader-based upscaling algorithms for video streaming. This helps you to apply multiple shaders sequentially to perform multi-pass upscaling and replaces the default [AMD FidelityFX Super Resolution 1.0](https://gpuopen.com/fidelityfx-superresolution/) shader. See {ref}`sec-custom-fragment-shader` for more information.<!--AC-2541-->

> * The Anbox Streaming SDK now uses the [Snapdragon™ Game Super Resolution(SGSR)](https://github.com/SnapdragonStudios/snapdragon-gsr) algorithm, which integrates upscaling and sharpening in one single GPU shader pass, by default.<!--AC-2543-->

Reviewer 2: IIRC we don't enable it by default. Should we change 'uses' to something more generic, like 'provides'?
```suggestion
* The Anbox Streaming SDK now provides the [Snapdragon™ Game Super Resolution(SGSR)](https://github.com/SnapdragonStudios/snapdragon-gsr) algorithm, which integrates upscaling and sharpening in one single GPU shader pass, by default.<!--AC-2543-->
```
@adglkh

Reviewer: Yes, I think the following sounds better.

The Anbox Streaming SDK includes two additional examples that demonstrate shader-based upscaling algorithms integrated with the JS SDK :
- [Snapdragon™ Game Super Resolution(SGSR)](https://github.com/SnapdragonStudios/snapdragon-gsr)
- [Anime4K high-quality real-time anime](https://github.com/bloc97/Anime4K)

### Other

* Android security updates for November 2024 (see [Android Security Bulletin - November 2024](https://source.android.com/docs/security/bulletin/2024-11-01) for more information).

> * The Android WebView has been updated to 130.0.6723.73.

Author: This is based on https://github.com/canonical/anbox/pull/369. Please confirm.

Reviewer 2: This is accurate.

## Removed functionality

Following deprecation in earlier releases, these functionalities have been removed from Anbox Cloud with the 1.24.0 release:

* Support for Ubuntu 20.04 (Focal Fossa)

> * Use of node controller to manage the port forwarding from an instance to expose a service

Author: @adglkh We need to discuss whether to remove all information about node controller across documentation or just this specific scenario.

Author: It is already in the deprecation notice since 1.23.0: https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/deprecation-notices/. Since it is not yet ready for removal, I will keep this in place until 1.25.0

## Deprecations

> < List deprecated features. Provide an alternative, if any, for the users to use instead of the deprecated features >

Author: Any deprecations this release?

Reviewer: Because of [this](https://github.com/canonical/anbox-cloud-docs/pull/212/files#r1831507950)
We need to mention the deprecation of the node controller in this section.

Author: Okay so we don't have to list it here for 1.24.0 because it is already in the deprecation notices ( See https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/deprecation-notices/). I will adjust the removal target release to 1.25.0 then.

## Known issues

> * Out of band support does not work as expected for Android 14 images because of a change in implicit intents feature in the upstream.<!--AC-2852-->

Reviewer 2: @adglkh Would you mind sharing the workaround you have found?

Reviewer: The argument above is not accurate. 

I think we can point people to the following page
https://developer.android.com/about/versions/14/behavior-changes-all#pending-broadcasts-queued
Specifically.
```
On Android 14, the system can [place context-registered broadcasts in a queue](https://developer.android.com/develop/background-work/background-tasks/broadcasts#android-14) while the app is in the [cached state](https://developer.android.com/guide/components/activities/process-lifecycle).
```
This is why the app didn’t receive the broadcast, even though it was sent out by the broadcast-data-channels-status action. This also means that any app or service intended to receive the com.canonical.anbox.BROADCAST_DATA_CHANNELS_STATUS broadcast must remain in the running state, not in a **cached** state. This is why our test case fails: the app wasn’t running at system startup.

So I don't think it's a known issue but a caveat if people use out of band v2 feature in Android 14

Author: @adglkh We have two options:

1. Drop the known issue as it is actually a caveat like you mentioned.
2. Include the change under Android section of features and improvements (yes, it is not necessarily a feature or improvement) since something has changed in the upstream behaviour and has caused a ripple effect in Anbox as well. We can list something like:
    > The out of band feature in Anbox Cloud works differently with Android 14 because of changes in upstream. On Android 14, the system can [place context-registered broadcasts in a queue](https://developer.android.com/develop/background-work/background-tasks/broadcasts#android-14) while the app is in the [cached state](https://developer.android.com/guide/components/activities/process-lifecycle). This means that an app needs to be running for it to receive the broadcast.

WDYT?

Reviewer: > @adglkh We have two options:
> 
> 1. Drop the known issue as it is actually a caveat like you mentioned.
> 2. Include the change under Android section of features and improvements (yes, it is not necessarily a feature or improvement) since something has changed in the upstream behaviour and has caused a ripple effect in Anbox as well. We can list something like:
>    > The out of band feature in Anbox Cloud works differently with Android 14 because of changes in upstream. On Android 14, the system can [place context-registered broadcasts in a queue](https://developer.android.com/develop/background-work/background-tasks/broadcasts#android-14) while the app is in the [cached state](https://developer.android.com/guide/components/activities/process-lifecycle). This means that an app needs to be running for it to receive the broadcast.
> 
> WDYT?

A: That sounds good to me. Let me reword this a bit
```
The out of band v2 feature in Anbox Cloud requires the Android app to be running to receive the broadcast If Android 14 image is in use due to the changed behavior that the system [place context-registered broadcasts in a queue]((https://developer.android.com/develop/background-work/background-tasks/broadcasts#android-14)) while the app is in the [cached state](https://developer.android.com/guide/components/activities/process-lifecycle).
```
Please adjust this when needed.

> * The following features are supported for AAOS images based on Android 14. This is not an intended support and will be removed in the next patch release:<!--AC-2848-->

Reviewer 2: It's not really 'supported' then, right? ;)
Also, it's not as good as I wrote it in the Jira story tbh: most of these are not working as intended, it's more of a partial, buggy support.
How should we phrase this? How about something along those lines:
```suggestion
* The following features can be enabled for AAOS images based on Android 14. This is not supported and may result in unexpected behavior. This will be removed in the next patch release:<!--AC-2848-->
```
Wdyt?

Reviewer: Sorry, I don't follow this. 
1. are we all agreeing to remove those features?
2. if so, are we going to do so in the next patch release?

Author: As far as I understand, these features should not be available for enabling but they are. We intend to remove these in the next patch release.
